app/uk/gov/hmrc/gform/repo/Repo.scala

```diff
@@ -18,107 +18,101 @@ package uk.gov.hmrc.gform.repo
 
 import cats.data.EitherT
 import cats.syntax.either._
+import com.mongodb.ReadPreference
+import org.bson.Document
+import org.mongodb.scala.bson.conversions.Bson
+import org.mongodb.scala.model.Filters._
+import org.mongodb.scala.model.{ Filters, FindOneAndReplaceOptions, IndexModel, InsertManyOptions }
 import play.api.libs.json._
-import reactivemongo.api.indexes.Index
-import reactivemongo.api.{ Cursor, DefaultDB, ReadConcern }
-import reactivemongo.bson.BSONObjectID
-
-import scala.concurrent.{ ExecutionContext, Future }
 import uk.gov.hmrc.gform.core.FOpt
 import uk.gov.hmrc.gform.exceptions.UnexpectedState
-import uk.gov.hmrc.mongo.ReactiveRepository
-import uk.gov.hmrc.play.http.logging.Mdc.preservingMdc
+import uk.gov.hmrc.mongo.MongoComponent
+import uk.gov.hmrc.mongo.play.json.PlayMongoRepository
+import org.mongodb.scala.model.Projections._
+import scala.concurrent.{ ExecutionContext, Future }
 
 class Repo[T: OWrites: Manifest](
   name: String,
-  mongo: () => DefaultDB,
+  mongoComponent: MongoComponent,
   idLens: T => String,
-  indexS: Seq[Index] = Seq.empty
+  indexes: Seq[IndexModel] = Seq.empty
 )(implicit
   formatT: Format[T],
   ec: ExecutionContext
-) extends ReactiveRepository[T, BSONObjectID](name, mongo, formatT) {
+) extends PlayMongoRepository[T](mongoComponent, name, formatT, indexes) {
   underlying =>
 
-  private val options = reactivemongo.api.QueryOpts(batchSizeN = Integer.MAX_VALUE)
-
-  import reactivemongo.play.json.ImplicitBSONHandlers._
-
-  override def indexes: Seq[Index] = indexS
-
-  def find(id: String): Future[Option[T]] = preservingMdc {
+  def find(id: String): Future[Option[T]] =
     underlying.collection
-      .find[JsObject, T](idSelector(id), None)
-      .one[T]
-  }
+      .find(equal("_id", id))
+      .first()
+      .toFutureOption()
 
-  def get(id: String): Future[T] = preservingMdc {
```

Comment:
How is `preservingMdc` functionality working with this PR?

Reply:
The new lib calls are MDC compatible. 

https://confluence.tools.tax.service.gov.uk/display/TEC/2021/03/04/HMRC+Mongo+is+now+available

See the "Why switch?" section